modules/vertice-cor-report/main.tf

```diff
@@ -0,0 +1,45 @@
+locals {
+  defaults_columns_for_selection = ["account_id", "action_type", "currency_code", "current_resource_details", "current_resource_summary",
+    "current_resource_type", "estimated_monthly_cost_after_discount", "estimated_monthly_cost_before_discount",
+    "estimated_monthly_savings_after_discount", "estimated_monthly_savings_before_discount",
+    "estimated_savings_percentage_after_discount", "estimated_savings_percentage_before_discount",
+    "implementation_effort", "last_refresh_timestamp", "recommendation_id", "recommendation_lookback_period_in_days",
+    "recommendation_source", "recommended_resource_details", "recommended_resource_summary", "recommended_resource_type",
+  "region", "resource_arn", "restart_needed", "rollback_possible", "tags"]
+  cor_columns_for_selection = length(var.cor_columns_for_selection) == 0 ? join(", ", local.defaults_columns_for_selection) : join(", ", setintersection(local.defaults_columns_for_selection, var.cor_columns_for_selection))
+}
+
+data "aws_s3_bucket" "vertice_cor_bucket" {
+  bucket = var.cor_report_bucket_name
+}
+
+resource "aws_bcmdataexports_export" "vertice_cor_report" {
+  export {
+    name = var.cor_report_name
+    data_query {
```

Comment:
The [provider example](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/bcmdataexports_export#example-usage) shows some more possible values for `data_query`:
```hcl
      table_configurations = {
        COST_AND_USAGE_REPORT = {
          TIME_GRANULARITY                      = "HOURLY",
          INCLUDE_RESOURCES                     = "FALSE",
          INCLUDE_MANUAL_DISCOUNT_COMPATIBILITY = "FALSE",
          INCLUDE_SPLIT_COST_ALLOCATION_DATA    = "FALSE",
        }
      }
```

Is that something we want here, or is it not needed?

Reply:
No, these options are available only for COST_AND_USAGE_REPORT table, we have here COST_OPTIMIZATION_RECOMMENDATIONS table, which has only these [options](https://docs.aws.amazon.com/cur/latest/userguide/table-dictionary-cor.html)